### DISCUSSION TITLE
Search Everywhere / Search Everything pop-up

### DISCUSSION THREAD

**Discussion by notpeter** [zed-industries/zed#24005] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

JetBrains users are accustomed to using `shift shift` to launch the "Search Everywhere" pop-up. It'd be nice if Zed had something similar

It searches the following:
- Types
- Filenames
- Symbols
- Actions
- Text (full text search)
- Menu items 
- Preferences (including inline toggles)

An initial version which included even just these would go a long way to providing a single entry-point so that users didn't have to remember "is it cmd+shift+p or cmd+p"...hmmm.
- Types
- Symbols
- Actions
- Filenames

<img width="670" alt="Screenshot 2024-07-11 at 15 44 52" src="https://github.com/zed-industries/zed/assets/145113/7b6426a5-9453-4287-90b0-ca0b50880b22">

**Comment by JosephTLyons** [comment]
Related:

- https://github.com/zed-industries/zed/issues/15896

**Comment by jonatino** [comment]
I'd also add making search a popup as well. I love that I don't need to leave my current tab/window when searching in intellij. I feel its much smoother and my workflow feels less interrupted when I just need to quickly search something. Intellij Search popup window is A++.

**Comment by jakubsuchybio** [comment]
I'd love "search everywhere" too, having separate "search project files" and "search project symbols" is kinda hard to get used to after switching from RustRover.

**Comment by bennidhamma** [comment]
I really want to use zed! It's weirdly fun rustifying as much of my toolchain as I can. but this feature is SUCH an instinct for me. To my potential detriment - I have less discipline about which files I put things because I just rely in the bot and the shift shift to basically get me places.

I am going to keep pushing myself to build my zed muscle, because rustification is a p0. but if I fail the lack of this feature may well be a reason. (and what on earth would happen to zed without me using it).

thanks for making a cool editor.

**Comment by dested** [comment]
I don't want to just spam discussions without adding value, but this is a really big one. Leaving the current editor for ephemeral actions like search everywhere or find all references is really annoying. I understand this is currently the workflow in editors like vscode, but it's a huge stretch coming from jetbrains tooling. Frankly its probably the number one thing keeping me stuck in webstorm given all the other functionality currently available in zed. Even support on how this could be implemented in a plugin would be appreciated!

**Comment by alexandru-wd** [comment]
Same. It is annoying to change the view context to search for a piece of text

**Comment by ibnIrshad** [comment]
For me this keybinding works well enough to get symbol/file search as long as you have a decent language server installed. I get Jetbrains-like search which is good enough for files and symbols, if not for everything else.

```
  {
    "context": "Workspace",
    "bindings": {
      "shift shift": "project_symbols::Toggle"
    }
  }
```

**Reply by MrHaroldA** [reply]
That does not find files (by filename) nor actions.

**Comment by MrHaroldA** [comment]
Every once in a while I will try Zed as "daily driver" instead of PhpStorm. Everything is faster, clearer, easier, more consistent ... but I really miss a "search everywhere" dialog, so I often switch back after just a few hours.

This missing feature really blocks me from using Zed full-time.

**Reply by lambda-science** [reply]
Agree with you @MrHaroldA same experience

**Comment by AWS-MattB** [comment]
Coming from JetBrains, I initially opted for the JetBrains keybindings, but I found that it didn't cover most of my daily-driver key bindings. So I switched back to the default keybindings and committed to learning Zed's defaults. That is a big investment due to all the rewiring of both the physical memory and the memory-memory associated with tasks. 

Having the ability to shift-shift to search Zed's list of actions would be very helpful for getting through the painful relearning process. Especially if the current keybindings for an action are displayed alongside the action.

**Comment by lambda-science** [comment]
Love this feature and would love to have

**Reply by lambda-science** [reply]
Basically it's combing search symbol and search file in one popup
